ducktape/cluster/remoteaccount.py

```diff
@@ -62,6 +63,8 @@ def _can_ping_url(self, url, headers):
     def ssh_command(self, cmd):
         if self.local:
             return cmd
+        if self.sudo_user is not None:
```

Comment:
@lindong28 Ok sure that makes sense. 

It sounds like we're converging on behavior where:
- `RemoteAccount` can be instantiated with a specified superuser (which defaults to root)
- ssh commands must specify  `sudo_user=True` to trigger this behavior